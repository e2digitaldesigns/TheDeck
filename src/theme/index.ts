export * from "./styles/forms.styles";
export * from "./styles/scrollBars.styles";
