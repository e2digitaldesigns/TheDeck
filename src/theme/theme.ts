export const theme = {
  sizes: {
    application: {
      height: "785px",
      width: "1280px"
    },
    header: {
      height: "2rem",
      width: "100vw"
    },
    macroDeck: {
      sideBar: { width: "240px" },
      mainContent: { width: "1016px" }
    }
  },
  colors: {
    body: "black",
    border: "#111",
    accent1: "#f542e9",
    accent2: "#291697",
    accent3: "#404040",
    accent4: "#2b2b2b",
    accent5: "#1a1a1a",
    gridBg: "43, 45, 48,",
    font: "white",
    main: "#100727",
    mainBackground: "black",
    scrollBackground: "#100727",
    scrollFont: "#dedede",
    scrollHeader: "black",
    scrollHeaderFont: "white"
  }
};
