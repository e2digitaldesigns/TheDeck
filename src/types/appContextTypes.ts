export interface IntAppDataIconSelector {
  isVisible: boolean;
  icon: string;
}

export interface IntAppDataActive {
  profileId: string;
  pageId: string;
  buttonPadId: string;
  actionId: string;
}

export interface IntDropZones {
  actionList: boolean;
  buttonPads: boolean;
  sideBarProfiles: boolean;
  sideBarStyles: boolean;
  styleHeader: boolean;
}

export interface IntAppContextInterface {
  iconSelector: IntAppDataIconSelector;
  active: IntAppDataActive;
  dropZones: IntDropZones;
}

export interface IntAppContextStateInterface {
  appState: IntAppContextInterface;
  setAppState: React.Dispatch<React.SetStateAction<any>>;
}

export type IntAppData = IntAppContextStateInterface;

export enum AppStateActive {
  ProfileId = "profileId",
  PageId = "pageId",
  ButtonPadId = "buttonPadId",
  ActionId = "actionId"
}
