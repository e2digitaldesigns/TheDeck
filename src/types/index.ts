/* istanbul ignore file */
export * from "./actionType";
export * from "./aplicationTypes";
export * from "./appContextTypes";
export * from "./buttonPadGridTypes";
export * from "./dragAndDropTypes";
export * from "./globalContextType";
export * from "./globalType";
export * from "./ipcRenderTypes";
export * from "./obsTypes";
