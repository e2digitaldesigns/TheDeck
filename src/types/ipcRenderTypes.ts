export enum IpcRendererTypes {
  database = "database",
  databaseReturn = "database:return",
  header = "header",
  importSettings = "importSettings",
  importSettingsReturn = "importSettings:return",
  mdPlaySound = "MD:playSound"
}

export enum IpcRendererTypesAction {
  close = "close",
  fsToggle = "fsToggle",
  minimize = "minimize",
  loadAppData = "loadAppData",
  saveAppData = "saveAppData"
}
