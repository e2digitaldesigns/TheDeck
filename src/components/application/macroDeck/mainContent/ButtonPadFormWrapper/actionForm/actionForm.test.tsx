import React from "react";
import "@testing-library/jest-dom";
import { cleanup, fireEvent, render } from "@testing-library/react";
import _cloneDeep from "lodash/cloneDeep";
import ActionForm from "./actionForm";
import { ThemeProvider } from "styled-components";
import { theme } from "../../../../../../theme/theme";

import initAppState from "../../../../../../__mock_data__/appState.json";
import initGlobalState from "../../../../../../__mock_data__/globalState.json";
import { IntAppData, IntGlobalData } from "../../../../../../types";

const mockAppState: IntAppData = {
  appState: _cloneDeep(initAppState),
  setAppState: jest.fn(appState => appState)
};

const mockGlobalState: IntGlobalData = {
  state: _cloneDeep(initGlobalState),
  setState: jest.fn(state => state)
};

const mockHooks = {
  useAppData: mockAppState,
  useGlobalData: mockGlobalState,
  useActions: {
    activateAction: jest.fn(),
    createAction: jest.fn(),
    deleteAction: jest.fn(),
    getAction: jest.fn(),
    getActions: jest.fn(),
    updateAction: jest.fn()
  },
  actionItem: mockGlobalState.state.actions[0]
};

jest.mock("../../../../../../hooks", () => ({
  useAppData: () => mockHooks.useAppData,
  useGlobalData: () => mockHooks.useGlobalData,
  useActions: () => ({
    activateAction: mockHooks.useActions.activateAction,
    createAction: mockHooks.useActions.createAction,
    getAction: mockHooks.useActions.getAction.mockImplementation(
      () => mockHooks.actionItem
    ),
    updateAction: mockHooks.useActions.updateAction,
    getActions: mockHooks.useActions.getActions.mockImplementation(() => [
      mockHooks.actionItem
    ]),
    deleteAction: mockHooks.useActions.deleteAction
  }),
  useObs: () => ({
    getScenes: () => [],
    getSources: () => []
  }),
  usePage: () => jest.fn(),

  useProfile: () => ({ readProfile: jest.fn() })
}));

const testSetup = () => {
  return render(
    <ThemeProvider theme={theme}>
      <ActionForm />
    </ThemeProvider>
  );
};

let wrapper: any = null;

beforeEach(() => {
  // wrapper = testSetup();
});

afterEach(() => {
  wrapper = null;
  mockAppState.appState = _cloneDeep(initAppState);
  mockGlobalState.state = _cloneDeep(initGlobalState);
  jest.resetAllMocks();
  jest.clearAllMocks();
  cleanup;
});

describe("<ActionFormwp/>", () => {
  xit("Should render without errors", () => {
    wrapper = testSetup();
    expect(wrapper).toMatchSnapshot();
  });

  xit("Should change action", () => {
    mockAppState.appState.active.actionId =
      mockGlobalState.state.actions[5]._id;
    wrapper = testSetup();
    const select = wrapper.getByTestId("action-form__action");
    expect(select).toBeTruthy();

    const actions = ["api", "delay", "exe", "sound", "keyTap", "md", "obs"];

    actions.forEach(action => {
      fireEvent.change(select, { target: { value: action } });
      expect(select.value).toBe(action);
    });
  });

  xit("Should change action", () => {
    mockAppState.appState.active.actionId =
      mockGlobalState.state.actions[0]._id;
    wrapper = testSetup();
  });
});
