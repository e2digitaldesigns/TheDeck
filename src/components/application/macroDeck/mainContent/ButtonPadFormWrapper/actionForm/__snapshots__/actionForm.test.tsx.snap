// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionFormwp/> Should render without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <fieldset
        class="sc-bdvvtL sc-jRQBWg Smgns"
      >
        <div>
          <label
            for="action"
          >
            Action:
          </label>
          <select
            class="sc-dkPtRN iRRDCl"
            data-testid="action-form__action"
            disabled=""
            name="action"
          >
            <option
              value="api"
            >
              API
            </option>
            <option
              value="delay"
            >
              Delay
            </option>
            <option
              value="exe"
            >
              Exe
            </option>
            <option
              value="keyTap"
            >
              KeyTap
            </option>
            <option
              value="md"
            >
              Macro Deck
            </option>
            <option
              value="obs"
            >
              OBS
            </option>
            <option
              value="sound"
            >
              Sound
            </option>
          </select>
        </div>
        <div>
          <label
            data-testid="form_field_selectField__label"
            for="md"
          >
            Md
            :
          </label>
          <select
            class="sc-dkPtRN iRRDCl"
            data-testid="form_field_selectField__input"
            name="subAction"
          >
            <option
              data-testid="form_field_selectField__option-choose"
              value=""
            >
              Choose
            </option>
            <option
              data-testid="form_field_selectField__option"
              value="mdHome"
            >
              mdHome
            </option>
            <option
              data-testid="form_field_selectField__option"
              value="mdPage"
            >
              mdPage
            </option>
            <option
              data-testid="form_field_selectField__option"
              value="mdProfile"
            >
              mdProfile
            </option>
            <option
              data-testid="form_field_selectField__option"
              value="mdProfileSelector"
            >
              mdProfileSelector
            </option>
            <option
              data-testid="form_field_selectField__option"
              value="mdReset"
            >
              mdReset
            </option>
            <option
              data-testid="form_field_selectField__option"
              value="mdSettings"
            >
              mdSettings
            </option>
          </select>
        </div>
      </fieldset>
      <button
        class="sc-hKwDye sc-gKclnd PLLBx gQFMTf"
        data-testid="action-form__submit"
        disabled=""
      >
        Submit
      </button>
    </div>
  </body>,
  "container": <div>
    <fieldset
      class="sc-bdvvtL sc-jRQBWg Smgns"
    >
      <div>
        <label
          for="action"
        >
          Action:
        </label>
        <select
          class="sc-dkPtRN iRRDCl"
          data-testid="action-form__action"
          disabled=""
          name="action"
        >
          <option
            value="api"
          >
            API
          </option>
          <option
            value="delay"
          >
            Delay
          </option>
          <option
            value="exe"
          >
            Exe
          </option>
          <option
            value="keyTap"
          >
            KeyTap
          </option>
          <option
            value="md"
          >
            Macro Deck
          </option>
          <option
            value="obs"
          >
            OBS
          </option>
          <option
            value="sound"
          >
            Sound
          </option>
        </select>
      </div>
      <div>
        <label
          data-testid="form_field_selectField__label"
          for="md"
        >
          Md
          :
        </label>
        <select
          class="sc-dkPtRN iRRDCl"
          data-testid="form_field_selectField__input"
          name="subAction"
        >
          <option
            data-testid="form_field_selectField__option-choose"
            value=""
          >
            Choose
          </option>
          <option
            data-testid="form_field_selectField__option"
            value="mdHome"
          >
            mdHome
          </option>
          <option
            data-testid="form_field_selectField__option"
            value="mdPage"
          >
            mdPage
          </option>
          <option
            data-testid="form_field_selectField__option"
            value="mdProfile"
          >
            mdProfile
          </option>
          <option
            data-testid="form_field_selectField__option"
            value="mdProfileSelector"
          >
            mdProfileSelector
          </option>
          <option
            data-testid="form_field_selectField__option"
            value="mdReset"
          >
            mdReset
          </option>
          <option
            data-testid="form_field_selectField__option"
            value="mdSettings"
          >
            mdSettings
          </option>
        </select>
      </div>
    </fieldset>
    <button
      class="sc-hKwDye sc-gKclnd PLLBx gQFMTf"
      data-testid="action-form__submit"
      disabled=""
    >
      Submit
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
